XiEditor/AppDelegate.swift

```diff
@@ -140,6 +140,25 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
         } 
         return applicationDirectory
     }()
+    
+    // The default name for XiEditor's error logs
+    let defaultCoreLogName = "xi_tmp.log"
+    
+    lazy var errorLogDirectory: URL? = {
+        let logDirectory = FileManager.default.urls(
+            for: .libraryDirectory,
+            in: .userDomainMask)
+            .first?
+            .appendingPathComponent("Logs")
+            .appendingPathComponent("XiEditor")
+
+        // create XiEditor log folder on first run
+        guard logDirectory != nil else { return nil }
+        try? FileManager.default.createDirectory(at: logDirectory!,
+                                                 withIntermediateDirectories: true,
+                                                 attributes: nil)
```

Comment:
Am I misreading this?  It seems like it swallows the createDirectory failure instead of resulting in nil.

Reply:
From what I understand:

Here, logDirectory is an URL? which will become nil if the user's /Library/ folder doesn't exist (I'm not sure exactly when, maybe when the admin disables it). `FileManager.default.createDirectory` returns void, as detailed in the documentation [here.](https://developer.apple.com/documentation/foundation/filemanager/1407884-createdirectory). I used `try?` here because we don't care if the creation fails, since as you and cmyr have mentioned, it shouldn't impede on normal operation even if it does.

Reply:
Since the intermediate directories arg is true, the only way for createDirectory to fail is that the directory doesn't exist.  I think the correct behaviour is that if creation fails it should set the logDirectory to nil rather than try to use it & fail later.